### PR TITLE
Feature/amazing traj 2

### PR DESCRIPTION
**Pull request by RotemKalisch**
Implement Sheevy's newest optimizations in CMF trajectory matrix and walk operations.

The new optimization works by changing the core algorithm of walk to walk as diagonally as possible instead of vertically.
We use trajectory matrices to calculate the diagonal steps, and changing the starting point accordingly between each steps.
Added a Position class as well to simplify calculations inside walk functions

Benchmark results (on my machine, take with a grain of salt):
image

Note that these benchmarks could not finish execution before these changes.

### CODE REVIEW

File: ramanujantools/cmf/cmf.py [hunk 1]
```diff
@@ -175,7 +175,37 @@ def simplify(self):
...
+        You should probably use `trajectory_matrix` instead.
+
+        Assumes trajectory is a simple diagonal - all abs values are at most 1
+        Args:
+            trajectory: a dict containing the amount of steps in each direction.
+            start: a dict representing the starting point of the multiplication.
+        Returns:
+            A matrix that represents a single step in the desired trajectory
+        """
+        if max([abs(value) for value in trajectory.values()], default=0) < 1:
```
Comment (ElyasheevLeibtag) [inline ramanujantools/cmf/cmf.py:192]: use position class: trajectory.longest

File: ramanujantools/cmf/cmf.py [hunk 2]
```diff
@@ -175,7 +175,37 @@ def simplify(self):
...
+            raise ValueError(
+                f"Called _calculate_diagonal_matrix with a trajectory that is not a simple diagonal: {trajectory}"
+            )
+        result = Matrix.eye(self.N())
+        position = start.copy()
+        for axis in self.axes_sorter(self.axes(), trajectory, start):
+            if trajectory[axis] == 0:
+                continue
+            sign = trajectory[axis] >= 0
+            result *= self.M(axis, sign).walk(self.axis_vector(axis, sign), 1, position)
```
Comment (ElyasheevLeibtag) [inline ramanujantools/cmf/cmf.py:202]: Use subs instead of walk

File: ramanujantools/cmf/cmf_benchmark.py [hunk 3]
```diff
@@ -21,6 +21,15 @@ def test_trajectory_matrix_euler(benchmark):
     benchmark(CMF.trajectory_matrix, cmf, trajectory, start)
```
Comment (ElyasheevLeibtag) [inline ramanujantools/cmf/cmf_benchmark.py:21]: lets try to test, 2f2 and 2f1 with deep trajectories, 12,13,14,20.

and also still was slow,
6f5 (z=1) at 1,1,1,1,1,1,2,2,2,2,2.

**Review by ElyasheevLeibtag — APPROVED** [review]
king

File: ramanujantools/cmf/cmf_benchmark.py (hunk 3 above)
Comment (RotemKalisch) [reply ramanujantools/cmf/cmf_benchmark.py:22]: @ElyasheevLeibtag I added the 2f2 deep trajectory test. Unfortunately loading a 6f5 CMF takes too long for now :(

**Pull request merged**